echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java

```diff
@@ -26,11 +29,22 @@
 @Component
 public class ExpectedArtifactExpressionEvaluationPostProcessor implements PipelinePostProcessor {
   private final ObjectMapper mapper;
-  private final ExpressionParser parser = new SpelExpressionParser();
+  private final ExpressionParser parser;
   private final ParserContext parserContext = new TemplateParserContext("${", "}");
 
-  public ExpectedArtifactExpressionEvaluationPostProcessor(ObjectMapper mapper) {
+  public ExpectedArtifactExpressionEvaluationPostProcessor(
+      ObjectMapper mapper, @Autowired ExpressionProperties expressionProperties) {
```

Comment:
Is `@Autowired` required here.  Doesn't everything get autowired via `@Component`?

Reply:
Yes, it is required as we are using it in constructor of a component and concluding its use inside the constructor.

Reply:
But...we don't have it for ObjectMapper.  Why is ExpressionProperties different?

Reply:
I verified without using `@Autowired`, build and tests are passing. So, updating code as suggested.